librad/src/git/identities/person.rs

```diff
@@ -189,6 +189,21 @@ where
     Ok(verified(storage).newer(a, b)?)
 }
 
+pub fn fast_forward<S>(
```

Comment:
I get how you're guided by the types, but you end up with a lot of IO:

* see if `known` exists
* load and verify `known`
* do ancestry check between `known` and `latest`
* update ref

There is also a correctness problem: the reason for introducing this is to reset `refs/namespaces/XXX/rad/id` to `refs/namespaces/YYY/rad/ids/XXX` _iff_ the latter can be fast-forwarded onto the former. _However_, this **must** be a fast-forward in the `git` sense, not the `tree` history sense, otherwise `YYY` can rewrite the history of `XXX`!

(No, we cannot actually prevent this entirely without a blockchain, but it is surely _safer_ to disallow history rewrites on identity branches wherever possible)

So I think this can be simplified to a function which takes a `VerifiedPerson` and:

1. Looks up the oid of the `rad/id`(ie. `refname_to_id(RefLike::try_from(pers.urn().with_path(None))?)`)
2. Calls `graph_descendant_of(pers.content_id, canonical_oid)`
3. If true, reset the ref from 1. to `pers.content_id`

Nb.: `VerifiedPerson` is already verified (qed), so if 1. and 2. pass we have also proven validity of the top-level person.

Reply:
Thanks for the help, that makes more sense. Hopefully, the simplified version is right :crossed_fingers: